src/_pytest/config/findpaths.py

```diff
@@ -22,44 +23,112 @@ def exists(path, ignore=OSError):
+    configuration, even if empty or without a "[pytest]" section.
+    """
+    iniconfig = _parse_ini_config(path)
+    if "pytest" in iniconfig:
+        return dict(iniconfig["pytest"].items())
+    else:
+        return {}
+
+
+def _get_ini_config_from_tox_ini(path: py.path.local) -> Optional[Dict[str, Any]]:
+    """Parses and validates a 'tox.ini' file for pytest configuration.
+
+    'tox.ini' files are only considered for pytest configuration if they contain a "[pytest]"
+    section.
+    """
+    iniconfig = _parse_ini_config(path)
+    if "pytest" in iniconfig:
+        return dict(iniconfig["pytest"].items())
+    else:
+        return None
+
+
+def _get_ini_config_from_setup_cfg(path: py.path.local) -> Optional[Dict[str, Any]]:
+    """Parses and validates a 'setup.cfg' file for pytest configuration.
+
+    'setup.cfg' files are only considered for pytest configuration if they contain a "[tool:pytest]"
+    section.
+
+    If a setup.cfg contains a "[pytest]" section, we raise a failure to indicate users that
+    plain "[pytest]" sections in setup.cfg files is no longer supported (#3086).
+    """
+    iniconfig = _parse_ini_config(path)
+
+    if "tool:pytest" in iniconfig.sections:
+        return dict(iniconfig["tool:pytest"].items())
+    elif "pytest" in iniconfig.sections:
+        fail(CFG_PYTEST_SECTION.format(filename="setup.cfg"), pytrace=False)
+    return None
+
+
+def _get_ini_config_from_pyproject_toml(
+    path: py.path.local,
+) -> Optional[Dict[str, Any]]:
+    """Parses and validates a 'setup.cfg' file for pytest configuration.
+
+    'setup.cfg' files are only considered for pytest configuration if they contain a "[tool:pytest]"
+    section.
+
+    If a setup.cfg contains a "[pytest]" section, we raise a failure to indicate users that
+    plain "[pytest]" sections in setup.cfg files is no longer supported (#3086).
+    """
+    import toml
+
+    config = toml.load(path)
+
+    result = config.get("tool", {}).get("pytest", {}).get("ini", None)
+    if result is not None:
+        # convert all scalar values to strings for compatibility with other ini formats
+        # conversion to actual useful values is made by Config._getini
+        def make_scalar(v):
+            return v if isinstance(v, (list, tuple)) else str(v)
+
+        return {k: make_scalar(v) for k, v in result.items()}
+    else:
+        return None
+
+
+def getcfg(args):
     """
     Search the list of arguments for a valid ini-file for pytest,
     and return a tuple of (rootdir, inifile, cfg-dict).
-
-    note: config is optional and used only to issue warnings explicitly (#2891).
     """
-    inibasenames = ["pytest.ini", "tox.ini", "setup.cfg"]
+    ini_names_and_handlers = [
+        ("pytest.ini", _get_ini_config_from_pytest_ini),
+        ("tox.ini", _get_ini_config_from_tox_ini),
+        ("setup.cfg", _get_ini_config_from_setup_cfg),
+        ("pyproject.toml", _get_ini_config_from_pyproject_toml),
```

Comment:
Shouldn't `pyproject.toml` have higher precedence? I know that people shouldn't have the configuration in multiple files, but they probably will.

Reply:
Hmm why should it have higher precedence?

Reply:
AFAIK it is meant to replace `setup.cfg` in the future, so I think it would be reasonable to put it right behind `pytest.ini`.

Reply:
Hmmm it makes sense, I suppose we want to make it the new "recommended" format right after pytest.ini.

I will make sure to note that order in the docs too.

Thanks!